test_http_server.py

```diff
@@ -24,28 +35,36 @@ def connection():
 @pytest.fixture
 def okresponse():
     now = formatdate(usegmt=True)
-    body = "Hello World, I need to travel."
+    body = "Thank you for the appropriate request."
     return ('HTTP/1.1 200 OK\r\n'
-        'Date: {}\r\nContent-Type: text/plain\r\nContent-Length: {}\r\n'
-        'Connection: close\r\n\r\n{}'.format(now, len(body), body))
+            'Date: {}\r\n'
+            'Content-Type: text/plain\r\n'
+            'Content-Length: {}\r\n'
+            'Connection: close\r\n\r\n{}'.format(now, len(body), body))
+
+
+@pytest.fixture
+def errorresponse():
+    now = formatdate(usegmt=True)
+    body = 'Method Not Allowed'
+    return ('HTTP 1.1 405 Method Not Allowed\r\n'
+            'Date: {}\r\n'
+            'Content-Type: text/plain\r\n'
+            'Content-Length: {}\r\n'
+            'Connection: close\r\n\r\n{}'.format(now, len(body), body)
+    )
 
 
 def test_response_ok(okresponse):
     assert response_ok() == okresponse
 
 
-def test_response_error():
-    now = formatdate(usegmt=True)
-    body = ("<html><body>\nThe server encountered an unexpected internal "
-    "error.</body></html>")
-    assert response_error() == ('HTTP 1.1 500 Internal Server Error'
-        '\r\nDate:{}\r\nContent-Type: text/html\r\nContent-Length: {}\r\n'
-        'Connection: close\r\n\r\n{}'.format(now, len(body), body)
-    )
+def test_response_error(errorresponse):
+    assert response_error('405') == errorresponse
```

Comment:
The fixtures you've created above that return an HTTP response, and how hard you've worked to make sure that that response is identical to the one that the server will return, should be an alert to you.  What does it prove to say that one bytestring you've typed out is identical to another one you've typed out and then sent through a socket?  

The test of your `response_ok` and `response_error` functions need to test the *nature* of the strings returned by those functions, not the content.  You want to make sure it is a well formed HTTP response, not that it is *this* well-formed HTTP response.  That's harder to do, but much more valuable.